staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -87,6 +93,17 @@ type authProxyRoundTripper struct {
 	rt http.RoundTripper
 }
 
+// NewCacheRoundTripper creates a roundtripper that reads the ETag on
+// response headers and send the If-None-Match header on subsequent
+// corresponding requests.
+func NewCacheRoundTripper(cacheDir string, rt http.RoundTripper) http.RoundTripper {
+	d := diskv.New(diskv.Options{BasePath: cacheDir})
+	t := httpcache.NewTransport(diskcache.NewWithDiskv(d))
```

Comment:
I think section 4.3.2 (Handling a Received Validation Request) applies to servers and proxies (and I think that's what they call "cache recipient" in your snippet). Then yes, if a proxy receives a request that is already cached, it should generate a 304 response, no doubt about that.

The relevant section, I think are 4.3.3 (Handling a Validation Response) and maybe 4.3.4 (Freshening Stored Responses upon Validation), but there is nothing specifically mentioning that.

I really doubt you should rely on an implementation detail for that.